src/ILVerify/tests/TestDataLoader.cs

```diff
@@ -0,0 +1,202 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Text;
+using Internal.IL;
+using Internal.TypeSystem.Ecma;
+using Newtonsoft.Json;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace ILVerify.Tests
+{
+    /// <summary>
+    /// Parses the methods in the test assemblies. 
+    /// It loads all assemblies from the test folder defined in <code>TestDataLoader.TESTASSEMBLYPATH</code>
+    /// This class feeds the xunit Theories
+    /// </summary>
+    class TestDataLoader
+    {
+        /// <summary>
+        /// The folder with the binaries which are compiled from the test driver IL Code
+        /// </summary>
+        public static string TESTASSEMBLYPATH = @"..\..\..\ILTests\";
```

Comment:
This looks very fragile. Is there any way this can be done in the build? The type system tests (\src\ILCompiler.TypeSystem\tests\TypeSystem.Tests.csproj) simply have a `ProjectReference` to the test data assemblies with a few flags set that get MSBuild to deploy the test artifacts right next to the test binary.